src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/EnqueueJobs.sql

```diff
@@ -36,6 +36,9 @@ BEGIN TRY
     IF @ForceOneActiveJobGroup = 1 AND EXISTS (SELECT * FROM dbo.JobQueue WHERE QueueType = @QueueType AND Status IN (0,1) AND (@GroupId IS NULL OR GroupId <> @GroupId))
       RAISERROR('There are other active job groups',18,127)
 
+    IF @GroupId IS NOT NULL AND EXISTS (SELECT * FROM dbo.JobQueue WHERE QueueType = @QueueType AND GroupId = @GroupId AND (Status = 4 OR CancelRequested = 1))
```

Comment:
Can we just check 

(SELECT * FROM dbo.JobQueue WHERE QueueType = @QueueType AND **JobId** = @**GroupId** AND CancelRequested = 1)

instead of full group check?